src/docker_config.rs

```diff
@@ -67,7 +68,14 @@ impl DockerCredentials {
             DockerCredentials::Split { username, password } => (username, password),
 
             DockerCredentials::Composite { auth } => {
-                String::from_utf8(general_purpose::STANDARD_NO_PAD.decode(auth)?)?
+                let decoded_auth =
+                    general_purpose::STANDARD_NO_PAD
+                        .decode(&auth)
+                        .or_else(|_| {
+                            info!("Attempting fallback decoding");
+                            general_purpose::STANDARD.decode(&auth)
+                        })?;
+                String::from_utf8(decoded_auth)?
```

Comment:
To draw direct attention to this from the PR comment, perhaps the default should just be `STANDARD` here instead rather than needing this fallback at all?

Reply:
let's either use `STANDARD_PAD` or `DecodePaddingMode::Indifferent`

Reply:
OK we just triple-checked and it turns out we need `STANDARD` (not `URL_SAFE`)